cardano-cli/src/Cardano/CLI/Shelley/Output.hs

```diff
@@ -5,32 +5,42 @@ module Cardano.CLI.Shelley.Output
   ( QueryTipOutput(..)
   , QueryTipLocalState(..)
   , QueryTipLocalStateOutput(..)
+  , ChainTipInfo(..)
   ) where
 
-import           Cardano.Api (AnyCardanoEra, ChainTip (..), EpochNo, serialiseToRawBytesHexText, EraHistory (..), CardanoMode)
+import           Cardano.Api (AnyCardanoEra, EpochNo, EraHistory (..), CardanoMode)
 import           Cardano.CLI.Shelley.Orphans ()
 import           Cardano.Prelude (Text)
 import           Cardano.Slotting.Block (BlockNo (..))
+import           Cardano.Slotting.Slot (SlotNo (..))
 import           Cardano.Slotting.Time (SystemStart (..))
 import           Control.Monad
 import           Data.Aeson (KeyValue, ToJSON (..), (.=))
 import           Data.Function (($), (.), id)
 import           Data.Maybe
 import           Data.Monoid (mconcat)
 import           Shelley.Spec.Ledger.Scripts ()
+import           Text.Show (Show)
 
 import qualified Data.Aeson as J
 import qualified Data.Aeson.Encoding as JE
 
+data ChainTipInfo = ChainTipInfo
```

Comment:
`ChainTipInfo` and `ChainTip` are extremely similar. Why can't we combine them or use one? Using `Maybe ChainTipInfo` isn't as informative. Does the `Nothing` case represent when the node's version does not support the query?

Reply:
`ChainTip` is defined like this:

```
data ChainTip = ChainTipAtGenesis
              | ChainTip !SlotNo !(Hash BlockHeader) !BlockNo
  deriving (Eq, Show)
```

This is the type returned by the chain sync protocol and is a pre-existing type.

```
getLocalChainTip :: LocalNodeConnectInfo mode -> IO ChainTip
```

In contrast, the local state query returns a `Point` which internally represents the hash as a `HeaderHash blk`.

`ChainTipInfo` is the new type and is defined like this:

```
data ChainTipInfo = ChainTipInfo
  { mBlockNo :: BlockNo
  , mSlotNo :: SlotNo
  , mHeaderHash :: Text
  } deriving Show
```

I introduced this type because the header hash is represented as `Text`.  I now how to convert a `HeaderHash blk` to `Text` and `Hash BlockHeader` to `Text`.

In order for me to use `ChainTip`, I need some way to convert `HeaderHash blk` to `Hash BlockHeader`.

Reply:
I think I may be able to use `ChainPoint` type.  Let me take a look.